### PR TITLE
whitelist external PRs on self-hosted

### PR DESCRIPTION
**Pull request by ab77**
tests:

* internal contribution (control) [here](https://github.com/balena-io/open-balena-vpn/actions/runs/5127328650/jobs/9222824010)
image

* external contribution [here](https://github.com/balena-io/open-balena-vpn/actions/runs/5127176594/jobs/9222770151#step:7:116).

image

### CODE REVIEW

File: flowzone.yml [hunk 1]
```diff
@@ -234,6 +234,25 @@
...
       fi
 
+  - &isApprovedExternalPullRequest
+    name: Check if an external PR is approved
+    id: is_approved_external_pr
+    env:
+      <<: *gitHubCliEnvironment
+    run: |
+      pr_labels="$(gh pr view ${{ github.event.pull_request.number }} --json labels -q .labels[].name)"
+      approve_label='gtg-${{ github.event.pull_request.head.sha || github.event.head_commit.id }}'
```
[the remark this answers is not in the document]
Comment (ab77) [reply flowzone.yml:244]: Yep, I'll update next week and try to wire it all up somehow to test.

File: flowzone.yml [hunk 2]
```diff
@@ -418,12 +437,12 @@
...
-        contains(fromJSON(inputs.runs_on), 'self-hosted') ||
-        contains(fromJSON(inputs.custom_runs_on), 'self-hosted') ||
-        contains(fromJSON(inputs.custom_runs_on).*, 'self-hosted') ||
-        contains(fromJSON(inputs.docker_runs_on).*, 'self-hosted')
+        contains(inputs.runs_on, 'self-hosted') ||
+        contains(inputs.custom_runs_on, 'self-hosted') ||
+        contains(inputs.docker_runs_on, 'self-hosted')
       )
     run: |
       echo "::error:External workflows can not be used with self-hosted runners. \
```
Comment (klutchell) [inline flowzone.yml:448]:
```suggestion
      echo "::error::External workflows can not be used with self-hosted runners. \
```
We should fix this typo I made previously. It's preventing the error from showing in the summary.

File: flowzone.yml [hunk 3]
```diff
@@ -396,7 +415,7 @@
       fi
 
   - &rejectExternalPullRequest
-    name: Reject external pull_request events
+    name: Reject external pull_request events on pull_request
```
Comment (klutchell) [inline flowzone.yml:418]: Good change

File: flowzone.yml [hunk 4]
```diff
@@ -406,7 +425,7 @@
       exit 1
 
   - &rejectInternalPullRequestTarget
-    name: Reject external pull_request events
+    name: Reject internal pull_request events on pull_request_target
```
Comment (klutchell) [inline flowzone.yml:428]: Good fix

File: flowzone.yml (hunk 2 above)
Comment (klutchell) [reply flowzone.yml:448]: Same typo is on line 423 and 433 if you could fix any instances of `::error:` with `::error::` that would be great

**Comment by ab77** [comment]
lgtm

**Comment by klutchell** [comment]
LGTM

**Pull request merged**